src/navigation/reroute.ts

```diff
@@ -194,17 +204,21 @@ export function reroute(
           return reroute(pendingPromises, eventArguments, true);
         }
 
-        const unloadPromises = appsToUnload.map(toUnloadPromise);
+        const unloadPromises: Promise<InternalApplication>[] =
```

Comment:
Once the `lifecycles` folder is migrated to typescript, the `unloadPromises` variable should have automatically-inferred types. Since that hasn't happened yet, though, I've put the types manually